tests/main/disconnect-undo/task.yaml

```diff
@@ -0,0 +1,35 @@
+summary: Test disconnect hook failure
+
+details: |
+  Test that failures of disconnect hook are handled gracefully and plug
+  remains connected.
+
+prepare: |
+  #shellcheck source=tests/lib/snaps.sh
+  . "$TESTSLIB/snaps.sh"
+  install_local test-disconnect
+
+restore: |
+  rm -f /var/snap/test-disconnect/common/do-not-fail
+
+execute: |
+  # sanity check, network is connected automatically
+  snap connections | MATCH "test-disconnect:network .*:network"
+
+  echo "Disconnect fails due to failing disconnect hook"
+  snap disconnect test-disconnect:network 2>&1 | MATCH "failure of disconnect hook"
+
+  echo "And network plug remains connected"
+  snap connections | MATCH "test-disconnect:network .*:network"
+
+  echo "Snap removal fails because of failing disconnect hook"
+  snap remove test-disconnect 2>&1 | MATCH "failure of disconnect hook"
+
+  echo "And the plug is still connected"
+  snap connections | MATCH "test-disconnect:network .*:network"
+
+  echo "Making disconnect hook happy"
+  touch /var/snap/test-disconnect/common/do-not-fail
```

Comment:
Interesting way out of this situation.

Reply:
Yeah, we drive hooks like this in other tests too.